frontend/src/components/controls/RisExpandableText.vue

```diff
@@ -0,0 +1,59 @@
+<script setup lang="ts">
+import { useElementId } from "@/composables/useElementId"
+import { ref, useTemplateRef, watchEffect } from "vue"
+
+const { length = 3 } = defineProps<{
+  /**
+   * Specifies the maximum number of visible lines.
+   * @default 3
+   */
+  length?: number
+}>()
+
+const expanded = defineModel<boolean>("expanded", { default: false })
+
+const canExpand = ref(false)
+
+const textContent = useTemplateRef("textContent")
+
+const { textId } = useElementId("expandable-text")
+
+watchEffect(() => {
+  if (textContent.value) {
+    canExpand.value =
+      textContent.value.scrollHeight > textContent.value.clientHeight
+  }
+})
+</script>
+
+<template>
+  <div>
+    <div
+      :id="textId"
+      ref="textContent"
+      :class="{ [$style.truncate]: !expanded }"
+    >
+      <slot />
+    </div>
+
+    <button
+      v-if="canExpand"
+      class="ris-link1-regular"
+      :aria-controls="textId"
+      :aria-expanded="expanded"
+      @click="expanded = !expanded"
+    >
+      <template v-if="expanded">Weniger anzeigen</template>
+      <template v-if="!expanded">Mehr anzeigen</template>
+    </button>
+  </div>
+</template>
+
+<style module>
+.truncate {
+  overflow: hidden;
+  display: -webkit-box;
+  -webkit-box-orient: vertical;
+  -webkit-line-clamp: v-bind(length);
```

Comment:
is this the only way to do that styling? according to mdn atleast using `-webkit-box-orient` is not only not a standard property but also deprecated: https://developer.mozilla.org/en-US/docs/Web/CSS/box-orient

Reply:
I took this from [this guide on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-line-clamp#truncating_a_paragraph), which is also how Tailwind implements it. I agree the syntax for line clamping is very awkward but as far as I know this is how you do it and how it works in all browsers 🥴